File: MayinTarlasi/MayinTarlasi.java
package mayintarlasi;

import java.util.Scanner;

public class MayinTarlasi {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Lütfen oyun alanının enini giriniz: ");
        int en = reader.nextInt();
        System.out.print("Lütfen oyun alanının boyunu giriniz: ");
        int boy = reader.nextInt();
        int puan = 0;
        int[][] tarla = new int[boy][en];
        for (int i = 0; i < boy; i++) {
            for (int j = 0; j < en; j++) {
                int bomba = (int) (Math.random() * 2);
                tarla[i][j] = bomba;
            }
        }
        System.out.print("Seçmek istediğiniz yerin koordinatlarını yazınız: ");
        for (;;) {
            int satir = reader.nextInt();
            int sutun = reader.nextInt();
            if (tarla[satir][sutun] == 1) {
                System.out.println("Game Over!!\nPuanınız : " + puan);
                break;
            }
            puan += 10;
            System.out.println("HALA HAYATTASIN...Devam\n10 puan kazanıldı");
            System.out.print("Hadi bir koordinat daha yaz: ");
        }
    }
}

File: MayinTarlasi/MayinTarlasi.py
import random


def mayin_tarlasi_hazirlama(en, boy):

    arena = [[0 for x in range(en)]for y in range(boy)]

    for y in range(boy):
        for x in range(en):
            rastgele = random.randint(0, 1)
            arena[y][x] = rastgele

    return arena


def oyuna_basla(arena):

    puan = 0
    oyun_sonu = False

    while oyun_sonu != True:

        koordinat_x = int(input("Seçmek istediğiniz x koordinatını girin : "))
        koordinat_y = int(input("Seçmek istediğiniz y koordinatını girin : "))

        if arena[koordinat_x][koordinat_y] == 0:
            puan += 10
            print("Hala hayattasın ve 10 puan kazandın")
        else:
            print("Boooom\nPuanın : {}".format(puan))
            oyun_sonu = True


oyuna_basla(mayin_tarlasi_hazirlama(int(input("En bilgisini girin : ")),int(input("Boy bilgisini girin : "))))